radis/api/kuruczapi.py

```diff
@@ -0,0 +1,466 @@
+
+        Args:
+            ielem (int): atomic number (e.g., Fe=26)
+            iion (int): ionized level (e.g., neutral=1, singly ionized=2, etc.)
+            df_ionE (pd.DataFrame): table of ionization energies
+
+        Returns:
+            ionE (float): ionization energy
+
+        Note:
+            NIST_iE is for now is .NIST_iE.txt
+        """
+
+        def f_droppare(x):
+            return (
+                x.str.replace("(", "", regex=True)
+                .str.replace(")", "", regex=True)
+                .str.replace("[", "", regex=True)
+                .str.replace("]", "", regex=True)
+                .str.replace("                                      ", "0", regex=True)
+            )
+
+        ionE = float(
+            f_droppare(
+                df_ionE[
+                    (df_ionE["At. num "] == ielem)
+                    & (df_ionE[" Ion Charge "] == iion - 1)
+                ]["      Ionization Energy (a) (eV)      "]
+            )
+        )
+        return ionE
+
+    def download_file(self):
+        """Download a file from an url to a specified output path."""
+
+        # extracts the file's name from l'URL
+        filename = self.url.split("/")[-1]
+
+        # Verify if the file exists already
+        if os.path.exists(filename):
+            print("File already exists, skipping download.")
+            return filename
+
+        with closing(requests.get(self.url, stream=True)) as r:
+            total_size = int(r.headers.get("content-length", 0))
+            block_size = 1024
+            with open(filename, "wb") as f, tqdm(
+                total=total_size,
+                unit="B",
+                unit_scale=True,
+                unit_divisor=1024,
+                desc=filename,
+            ) as pbar:
+                for data in r.iter_content(block_size):
+                    f.write(data)
+                    pbar.update(len(data))
+        return filename
+
+    def Sij0(self, A, g, nu_lines, elower, QTref):
+        # The int column of the df can be computed using this method but the default option option rather uses Radis linestrength_from_Einstein
+        """Reference Line Strength in Tref=296K, S0.
+
+        Note:
+        Tref=296K
+
+        Args:
+        A: Einstein coefficient (s-1)
+        g: the upper state statistical weight
+        nu_lines: line center wavenumber (cm-1)
+        elower: elower
+        QTref: partition function Q(Tref)
+        Mmol: molecular mass (normalized by m_u)
+
+        Returns:
+        Sij(T): Line strength (cm)
+        """
+        hcperk = 1.4387773538277202  # hc/kB (cm K)
+        ccgs = 29979245800.0
+        Tref = 296.0
+        S0 = (
+            -A
+            * g
+            * np.exp(-hcperk * elower / Tref)
+            * np.expm1(-hcperk * nu_lines / Tref)
+            / (8.0 * np.pi * ccgs * nu_lines**2 * QTref)
+        )
+        return S0
+
+    def read_kurucz(self, kuruczf):
```

Comment:
Same, reference the author in the docstrings. Best is even to cite Exojax paper DOI